### DISCUSSION TITLE
Loading problem with jnlp

### DISCUSSION THREAD

**Discussion by tstegen** [ESAPI/esapi-java-legacy#849] [Q&A]
I am using the ESAPI library in my project, but I am having trouble loading the properties from the resources. The solution downloads a client via JNLP and is unable to access the esapi.properties file from the classpath, even though it is present in the resources (src/resources/esapi/esapi.properties). It seems the file might not be at the root of the JAR file for JNLP.

the jnlp contains :
```
<resources>
 <jar>core.jar</jar> (with esapi.propreties)
 <jar>other.jar</jar>
</resources>
```

**Comment by kwwall** [comment]
This is not an answer or even an attempted answer, but rather a follow up question.

Do you mean JNLP is used to launch applets in browsers? Are you running your application as a standalone desktop application? I didn't even know applets were still supported, but given this https://docs.oracle.com/javase/tutorial/deployment/deploymentInDepth/jnlp.html has a copyright of 1995 and 2022, maybe they are. News to me.

Regardless, I think if you want help with this, you are going to have to be a LOT more specific with details, and most likely give us a small 'hello world' like example and tell us how you have it configured (e.g., using Java Web Start or what?) and how it gets launched so that we can reproduce it in some simple example if you want help. Especially if you are doing this as an applet in a browser, which I haven't worked on since the early 2000s. For the ESAPI example, you can just use ESAPI's Logger to print something or do a simple Validator example. Pretty much anything that can be used to confirm if ESAPI.properties is loading correctly. But otherwise, I honestly don't think we have enough information to assist you.

**Comment by planetlevel** [comment]
JNLP launches client side Java programs.  I think you’re on the right track getting ESAPI properties on the classpath.  See https://www.javadoc.io/doc/org.owasp.esapi/esapi/2.0.1/org/owasp/esapi/reference/DefaultSecurityConfiguration.htmlCan you add an “esapi” or “.esapi” directory to your classpath with the esapi.properties in it?  Or maybe you could add “src/resources” to your classpath.—JeffOn Jul 20, 2024, at 3:57 PM, Kevin W. Wall ***@***.***> wrote:﻿
This is not an answer or even an attempted answer, but rather a follow up question.
Do you mean JNLP is used to launch applets in browsers? Are you running your application as a standalone desktop application? I didn't even know applets were still supported, but given this https://docs.oracle.com/javase/tutorial/deployment/deploymentInDepth/jnlp.html has a copyright of 1995 and 2022, maybe they are. News to me.
Regardless, I think if you want help with this, you are going to have to be a LOT more specific with details, and most likely give us a small 'hello world' like example and tell us how you have it configured (e.g., using Java Web Start or what?) and how it gets launched so that we can reproduce it in some simple example if you want help. Especially if you are doing this as an applet in a browser, which I haven't worked on since the early 2000s. For the ESAPI example, you can just use ESAPI's Logger to print something or do a simple Validator example. Pretty much anything that can be used to confirm if ESAPI.properties is loading correctly. But otherwise, I honestly don't think we have enough information to assist you.

**Comment by tstegen** [comment]
Hi @kwwall ,

Thank you for your response. I wanted to clarify that the application is launched as a standalone desktop application via OpenWebStart, not as an applet in a browser. Upon investigating the downloaded application file, I found that the resources were not included, hence they were not in the classpath.

After further research, I discovered that an assembly issue was causing this problem. I apologize for any inconvenience caused by my earlier frustration after spending several hours on unsuccessful attempts to resolve this.

Thank you for your patience and understanding.
